FilteringApp/FilteringApp.Core/Utils/ArrayUtils.cs

```diff
@@ -39,37 +40,33 @@ public static int AddArrayElements(int[] array)
             return theSum;
         }
 
-        public static ArrayParseResult CheckArray(string[] stringArray)
+        public static int[] CheckArray(this string[] stringArray)
         {
             var array = new int[stringArray.Length];
+
             for (int i = 0; i < array.Length; i++)
             {
                 if (!int.TryParse(stringArray[i], out var value))
                 {
-                    var error = $"Wrong Input in array: {stringArray[i]} Postion :{i + 1}";
-                    return new ArrayParseResult(error);
+                    throw new ArgumentException($"Wrong Input in array: {stringArray[i]} Postion :{i + 1}");
                 }
-                else
-                {
                     array[i] = value;
-                }
+                
             }
-
-            return new ArrayParseResult(array);
+            return array;
         }
 
-        public static UserValueParseResult CheckFilteringValue(string value)
+        public static int CheckFilteringValue(this string value)
         {
             var userInput = value;
```

Comment:
Redundant line

Reply:
I am having difficulty finding this.

Reply:
I was talking about the line
`var userInput = value;`

You don't need extra variable here

Reply:
right , done

Reply:
I can still see it is there :)